src/python_testing/execute_python_tests.py

```diff
@@ -0,0 +1,92 @@
+#
+#    Copyright (c) 2024 Project CHIP Authors
+#    All rights reserved.
+#
+#    Licensed under the Apache License, Version 2.0 (the "License");
+#    you may not use this file except in compliance with the License.
+#    You may obtain a copy of the License at
+#
+#        http://www.apache.org/licenses/LICENSE-2.0
+#
+#    Unless required by applicable law or agreed to in writing, software
+#    distributed under the License is distributed on an "AS IS" BASIS,
+#    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#    See the License for the specific language governing permissions and
+#    limitations under the License.
+#
+
+import glob
+import os
+import subprocess
+
+# Function to load environment variables from a YAML file
+
+
+def load_env_from_yaml(file_path):
+    with open(file_path, 'r') as file:
+        for line in file:
+            if line.strip():  # Skip empty lines
+                key, value = line.strip().split(': ', 1)
+                os.environ[key] = value
+
+
+# Load environment variables from test_env.yaml
+load_env_from_yaml('/tmp/test_env.yaml')
+
+# Define the base command to run tests
+base_command = "./scripts/tests/run_python_test.py"
+
+# Define the directory to search for Python scripts
+search_directory = "src/python_testing"
+
+# Define the files and patterns to exclude
+excluded_patterns = [
+    "MinimalRepresentation.py",
+    "TC_ACL_2_2.py",
+    "TC_CNET_4_4.py",
+    "TC_DGGEN_3_2.py",
```

Comment:
why are these TC_* excluded?

We need some explanation - I believe it may have to do with test binaries not being able to support them, however it would be nice to have a comment for each (or for a group like the support/test ones) explaining the reasoning.

@cecille do you happen to have a TLDR for some of them?

Reply:
For why I excluded: I just excluded the list of tests from the tests.yaml block which were not the app tests or being run through that block of tests mentioned.